Backpack/src/main/java/net/skyscanner/backpack/contentbubble/BpkContentBubble.kt

```diff
@@ -0,0 +1,224 @@
+package net.skyscanner.backpack.contentbubble
+
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.os.Build
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.use
+
+/**
+ * [BpkContentBubble] is designed to render a single item inside a "bubble".
+ * Think of message apps and how they show content inside text bubbles.
+ *
+ * No background or padding should be set directly to this view, set it directly its
+ * child.
+ */
+open class BpkContentBubble @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0,
+  private val pointerDrawable: Drawable = ContextCompat.getDrawable(context, R.drawable.content_bubble_default_pointer)!!,
+  private val cornerRadiusDrawable: Drawable = ContextCompat.getDrawable(context, R.drawable.content_bubble_default_radii)!!
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+  enum class PointerPosition(internal val id: Int, internal val offset: Float) {
+    START(0, 0.25f),
+    MIDDLE(1, 0.5f),
+    END(2, 0.75f)
+  }
+
+  private lateinit var mask: Bitmap
+  private lateinit var maskCanvas: Canvas
+
+  private lateinit var radiusMask: Bitmap
+  private lateinit var radiusMaskCanvas: Canvas
```

Comment:
The bitmaps could be store statically with lazy initialisations in get(Context) methods.